### PR TITLE
[WIP] fix Windows build

### LINKED ISSUE #1622: Division by zero in Windows builds
## 🐛 Bug

With the introduction of the new metrics API, there are issues with division by zero.

### To Reproduce

To reproduce, run the most minimal `fairseq-train` on a Windows machine. The error is also seen in the test failures from [Windows CI](https://github.com/pytorch/fairseq/pull/1595/checks?check_run_id=390286089)

#### Code sample

N/A

### Expected behavior

There shouldn't be division by zero errors.

### Environment

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0) 1.3.1
 - OS (e.g., Linux): Windows
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): See the build script.
 - Python version: 3.6.7
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information: N/A

### Additional context

Strangely, this doesn't seem to happen in Unix builds...

### PR DESCRIPTION
**Pull request by erip**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes #1622

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by erip** [comment]
052bacbf seems to have fixed the division-by-zero errors. There are some other failures related to Pytorch runtime errors -- this is likely some weird installation artifact.

File: examples/speech_recognition/infer.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def main(args):
     # Load ensemble
     logger.info("| loading model(s) from {}".format(args.path))
     models, criterions, _model_args = load_models_and_criterions(
-        args.path.split(":"),
+        args.path.split("::"),
```
Comment (myleott) [inline examples/speech_recognition/infer.py]: Can you replace this with `os.pathsep` instead: https://docs.python.org/3/library/os.html#os.pathsep

File: examples/nonautoregressive_translation/README.md [hunk 2]
```diff
@@ -110,7 +110,7 @@ fairseq-generate \
     data-bin/wmt14_en_de_distill \
     --gen-subset test \
     --task translation_lev \
-    --path checkpoints/checkpoint_best.pt:at_checkpoints/checkpoint_best.pt \
+    --path checkpoints/checkpoint_best.pt::at_checkpoints/checkpoint_best.pt \
```
Comment (myleott) [inline examples/nonautoregressive_translation/README.md]: Let's revert back to `:` and use `os.pathsep` when parsing.

Comment (erip) [reply examples/nonautoregressive_translation/README.md]: @myleott done!

**Comment by erip** [comment]
There seems to be a hodgepodge of other failures here ranging from Pytorch type mismatches (which are likely some weird platform dependent implicit conversions?) to JSON parsing errors. I'll be addressing these in stages and can hopefully have this wrapped up soon. The last commit I'd like to make as a part of this series of fixes is to remove `fail-fast` from CI so every platform is in lock-step.

**Comment by erip** [comment]
Looks like there are a few lingering issues:

- `DataLoader` seems to be crashing on Windows - my research hasn't led me to a conclusion about whether this is a resolved issue or remains in pytorch...
- Reproducibility tests are not passing - I haven't tracked this down, but it seems like the logging order is different between unix and Windows. I suspect this has something to do with the `rename_logger` function, but haven't pushed it much further than that.
- Some odds and ends stuff which I haven't explored.

**Comment by erip** [comment]
I've dug a bit deeper. I think some of the `DataLoader` issues may be related to the reproducibility tests: the new [logging logic doesn't work with `TemporaryDirectory`s](https://stackoverflow.com/questions/59833204/why-cant-my-file-handlers-file-within-a-temporary-directory-be-cleaned-up-on-w/). I think if it gets refactored in some smart way, both could be solved.

**Comment by erip** [comment]
@myleott In my judgment, the above commits will yield something that puts fairseq in a good place on Windows. It would be good to fix the tests, but that seems to be a much more involved process. Do you think I should keep this as a WIP or would you like to do this more piecewise (merge the above commits and open a separate PR for the remaining TODOs)?

File: fairseq/tasks/language_modeling.py [hunk 3]
```diff
@@ -154,7 +154,7 @@ def load_dataset(self, split, epoch=0, combine=False, **kwargs):
         Args:
             split (str): name of the split (e.g., train, valid, test)
         """
-        paths = self.args.data.split(":")
+        paths = self.args.data.split("::")
```
Comment (myleott) [inline fairseq/tasks/language_modeling.py]: Change to os.pathsep

**Comment by myleott** [comment]
FYI, I also moved the windows build into its own workflow, so it doesn't affect the badge in the README. We will add it back into the main workflow once it's passing :)

**Pull request closed without merging**